pkg/machines/libvirt-common.js

```diff
@@ -884,6 +884,64 @@ export function unknownConnectionName(action, libvirtServiceName) {
     };
 }
 
+export function updateDiskReadOnly(domXml, diskTarget) {
+    const domainElem = getDomainElem(domXml);
+    if (!domainElem)
+        throw new Error("updateDiskReadOnly: domXML has no domain element");
```

Comment:
To be honest I put it there only because I have noticed it in other updateXML functions, so I thought there is reason for that. But now when I read about it a bit, libvirt would not allow this scenario, and since we pull all the XML for updateXML functions from libvirt, I guess it's safe not to have it.